Fix MALLOC_HEAP_BREAKDOWN builds

Scony opened the pull request:
#### a235408c2b4eb12216d519e996f70828b9a45e19
<pre>
Fix MALLOC_HEAP_BREAKDOWN builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=301074">https://bugs.webkit.org/show_bug.cgi?id=301074</a>

Reviewed by Fujii Hironori.

This change fixes compilation when MALLOC_HEAP_BREAKDOWN is used.

Canonical link: <a href="https://commits.webkit.org/301839@main">https://commits.webkit.org/301839@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/00891e7270c725ebdf11881b8f93e018aa444578

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127140 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46775 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37907 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134204 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78695 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/129011 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47388 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55301 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96754 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64789 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130088 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37936 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113867 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77258 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36807 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31983 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77584 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107805 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32348 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136684 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53792 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41439 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105275 "44 flakes 79 failures") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54300 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110222 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104961 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50487 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28922 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51348 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19904 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53723 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59709 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52955 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56290 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54718 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/inspector/InspectorThreadableLoaderClient.h:
```diff
@@ -39,6 +39,8 @@ class ThreadableLoader;
 
 namespace Inspector {
 
+DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(Loader);
```
Use the actual class name.
~~~
DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(InspectorThreadableLoaderClient);
~~~

Scony replied:
I had to adjust the heap name for that, but as this is separate namespace from `Loader`, I think that was the original's author intent - to make it separate.

fujii commented on Source/WebCore/inspector/InspectorThreadableLoaderClient.h:
```diff
@@ -39,9 +39,11 @@ class ThreadableLoader;
 class InspectorThreadableLoaderClient final : public WebCore::ThreadableLoaderClient {
     WTF_MAKE_NONCOPYABLE(InspectorThreadableLoaderClient);
-    WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(InspectorThreadableLoaderClient, Loader);
+    WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(InspectorThreadableLoaderClient, InspectorThreadableLoaderClient);
```
Ah, sorry. I didn't understand the problem. It seems that we should add `#include "LoaderMalloc.h"` in InspectorThreadableLoaderClient.h like DocumentThreadableLoader.h does.
https://github.com/WebKit/WebKit/blob/78f2de6b87656ba70ba69b71ae948a6e5283871d/Source/WebCore/loader/DocumentThreadableLoader.h#L36

Scony replied:
It's already included, but the declaration is in parent namespace. I've added `using` to make it working and reporting to `Loader` - let me know what you think.

fujii commented on Source/WebCore/inspector/InspectorThreadableLoaderClient.h:
```diff
@@ -39,6 +39,10 @@ class ThreadableLoader;
+#if ENABLE(MALLOC_HEAP_BREAKDOWN)
+using WebCore::LoaderMalloc;
+#endif
+
```
Umm, this doesn't look nice to me. No other code does this tricky hack. I think the second patch looks better. WDYT?
https://github.com/WebKit/WebKit/commit/310a7fbf859a335d18851f31a18e38ed69b16287

Scony replied:
I also think so.

Scony committed:
Fix MALLOC_HEAP_BREAKDOWN builds
https://bugs.webkit.org/show_bug.cgi?id=301074

Reviewed by Fujii Hironori.

This change fixes compilation when MALLOC_HEAP_BREAKDOWN is used.

Canonical link: https://commits.webkit.org/301839@main
